test/lib/util/errata_test.rb

```diff
@@ -36,29 +36,60 @@ def setup
+      # The same goes for the number of module streams that the single RPM belongs to.
+      assert_equal 1, @modular_erratum.packages.count
+      assert_equal 1, @modular_erratum_package.module_streams.count
+      assert_equal [@erratum], filter_errata_by_pulp_href([@modular_erratum, @erratum],
+                                                          [@fedora_one.pulp_id, @fedora_two.pulp_id, @modular_rpm.pulp_id],
+                                                          [@fedora_one.filename, @fedora_two.filename, @modular_erratum_package.filename],
+                                                          [@module_stream.module_spec])
+    end
+
+    # https://projects.theforeman.org/issues/35614
+    def test_filter_by_pulp_href_includes_errata_with_proper_module_streams
+      # If the erratum's modular RPM count is no longer 1, the test no longer makes sense.
+      # Ensure that all modular packages are accounted for in the final assertion.
+      # The same goes for the number of module streams that the single RPM belongs to.
+      assert_equal 1, @modular_erratum.packages.count
+      assert_equal 1, @modular_erratum_package.module_streams.count
+      assert_equal [@erratum, @modular_erratum].sort, filter_errata_by_pulp_href([@modular_erratum, @erratum],
+                                                          [@fedora_one.pulp_id, @fedora_two.pulp_id, @modular_rpm.pulp_id, @module_stream.pulp_id],
+                                                          [@fedora_one.filename, @fedora_two.filename, @modular_erratum_package.filename],
+                                                          [@module_stream.module_spec])
     end
 
     def test_filter_by_pulp_id_returns_nothing_with_empty_list_of_pulp_ids
-      assert_empty filter_errata_by_pulp_href([@erratum, @erratum2], [], [@fedora_one.filename, @fedora_two.filename])
+      assert_empty filter_errata_by_pulp_href([@erratum, @erratum2], [], [@fedora_one.filename, @fedora_two.filename], [])
     end
 
     def test_filter_by_pulp_id_returns_nothing_with_empty_list_of_errata
-      assert_empty filter_errata_by_pulp_href([], [@fedora_one.pulp_id], [@fedora_one.filename, @fedora_two.filename])
+      assert_empty filter_errata_by_pulp_href([], [@fedora_one.pulp_id], [@fedora_one.filename, @fedora_two.filename], [])
     end
 
-    # "Proper" errata here means errata with no RPMs that exist in its source repo any empty errata.
+    # "Proper" errata here means errata with no RPMs that exist in its source repo and any empty errata.
     def test_filter_by_pulp_id_returns_proper_errata_if_no_matching_packages
-      assert_equal [@erratum2, @empty_erratum].sort, filter_errata_by_pulp_href([@erratum, @erratum2, @empty_erratum], ["floop"], [@fedora_one.filename, @fedora_two.filename]).sort
+      assert_equal [@erratum2, @empty_erratum].sort, filter_errata_by_pulp_href([@erratum, @erratum2, @empty_erratum], ["floop"], [@fedora_one.filename, @fedora_two.filename], []).sort
```

Comment:
We had some issues with array sorts on activerecords not being consistent. I am not sure if that applies here.
https://github.com/Katello/katello/pull/10341#issuecomment-1310844946 but it might be better to use order_by(:id) just in case this introduces ambiguity.

Reply:
the `order` method only works on active record objects rather than arrays.  I did a quick test and the ordering seemed to be consistent for errata.